### DISCUSSION TITLE
larger compass?

### DISCUSSION THREAD

**Discussion by halset** [maplibre/maplibre-native#727] [Q&A]
Is there a way to make the compass larger? In Mapbox GL Native, that was possible by setting the compassView.image property to a larger compass image, but they remove that functionality. What is the best way to do this with maplibre-gl-native?

I have tried setting the compassView.image to a larger image and calling sizeToFit on the compassView, but the compass does not change.

**Comment by MarkusMannheim** [comment]
Good question; I'd love to know, the answer, too.

**Comment by halset** [comment]
In my app, I ended up creating a separate custom compass UIView and redraw it with core graphics when the maps rotates. I also ended up having a custom gesture recognizer to handle tap for switching between north up and course up. 

So perhaps it is just fine to have a basic compass in maplibre-gl-native and then app developers can make their own if they want to. Just changing the image to a larger one would have been simpler than writing a custom compass.

**Reply by boldtrn** [reply]
If this is not possible anymore, maybe you could contribute this feature back to maplibre-gl-native?

**Comment by ovivoda** [comment]
@halset have you tried `setCompassMargins` from `UiSettings` class?
